include/wasm/WasmCommon.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
```

Comment:
This file is auto-generated by CMake, I am not sure whether to track it or not.

Reply:
The usual practice for auto-generated files is to generate them in the build directory, and never modify the input source tree. It's especially useful when you set up multi-config builds which might need different versions of the same headers. I think faasm already breaks this rule with the current protobuf configuration (I did change that in my fork at some point), but it might be good to follow for future changes

Reply:
Great, thanks for chipping in, was about to ask you at some point.

We could eventually change the protobuf headers as well.

Reply:
Follow-up, I realised `FAASM_INCLUDE_DIR` points to `CMAKE_LIST_DIR/include`, should we eventually move to `CMAKE_BUILD_DIR/include` and copy the includes there?

I will now do it just for one file, but for the protobuf change for example.

Reply:
I think both should be there: `CMAKE_LIST_DIR/include` first, and then `CMAKE_BUILD_DIR/include` second. That way the main source tree files take precedence, but you still get access to generated headers. It also means you can theoretically alias names and use `#include_next` to get an identically named header from the build tree if you want insert a hook/wrap the generated header in some extra code

Reply:
@eigenraven requesting a review from you as well see if i have overlooked something